File: .tmp_py4013/batch_18_dump/3979_maximum_valid_pair_sum.js
// LeetCode 3979 - Maximum Valid Pair Sum
// https://leetcode.com/problems/maximum-valid-pair-sum/
var maxValidPairSum = function(nums, k) {
        let ans = 0, x = 0;
        for (let j = k; j < nums.length; j++) {
            let y = nums[j];
            x = Math.max(x, nums[j - k]);
            ans = Math.max(ans, x + y);
        }
        return ans;
    
};

File: .tmp_rs4013/rust_check22/3979_maximum_valid_pair_sum.rs
struct Solution;
// LeetCode 3979 - Maximum Valid Pair Sum
// https://leetcode.com/problems/maximum-valid-pair-sum/

impl Solution {
    pub fn max_valid_pair_sum(nums: Vec<i32>, k: i32) -> i32 {
        let mut ans = 0;
        let mut x = 0;
        for j in k as usize..nums.len() {
            let y = nums[j];
            x = x.max(nums[j - k as usize]);
            ans = ans.max(x + y);
        }
        ans
    }
}

fn main() {}
